data_io.c

```diff
@@ -151,7 +152,9 @@ void data_io_rom_upload(char *rname, char mode) {
   user_io_change_into_core_dir();
 
   strcpy(s, "        ROM");
-  strncpy(s, rname, strlen(rname));
+  for (i = 0; i < (strlen(rname) < 8 ? strlen(rname) : 8); i++) {
+    s[i] = rname[i];
+  }
```

Comment:
What about:
``` 
int l = strlen(rname) < 8 ? strlen(rname) : 8;
strncpy(s, rname, l)
```

Reply:
I think I tried that and gcc still complained...

Yes, this works, even:
```
strncpy(s, rname, strlen(rname) < 8 ? strlen(rname) : 8);
```

Tried to reproduce the warning. I think I tried
```
strncpy(s, rname, strlen(rname) <= 8 ? strlen(rname) : 8);
```
first and this doesn't work.